h2/src/main/org/h2/mode/FunctionsOracle.java

```diff
@@ -128,10 +142,40 @@ protected Value getValueWithArgs(SessionLocal session, Expression[] args) {
         case TO_TIMESTAMP_TZ:
             result = ToDateParser.toTimestampTz(session, v0.getString(), v1 == null ? null : v1.getString());
             break;
+        case REGEXP_REPLACE:
+            Boolean replaceBackslashRefs = session.getMode().regexpReplaceBackslashReferences;
+            String regexpResult = regexpReplace(replaceBackslashRefs, v0, v1, v2, v3, v4, v5);
+            result = ValueVarchar.get(regexpResult, session);
+            break;
         default:
             throw DbException.throwInternalError("type=" + info.type);
         }
         return result;
     }
 
+    private static String regexpReplace(Boolean replaceBackslashRefs, Value ... args) {
+        String input = args[0].getString();
+        String regexp = args[1].getValueType() != Value.NULL ? args[1].getString() : "";
+        String replacement = args[2].getValueType() != Value.NULL ? args[2].getString() : "";
+        if (replaceBackslashRefs) {
+            replacement = StringUtils.replaceBackslashReferences(replacement);
+        }
+        int position = 1;
+        int occurrence = 0;
+        String regexpMode = "";
+
+        if (args[3] != null) {
+            if (args[3].getValueType() == Value.VARCHAR) {
+                regexpMode = args[3].getString() == null ? args[3].getString() : "";
+            } else if (args[3].getValueType() == Value.INTEGER) {
```

Comment:
What are you trying to do here? Fourth argument may only be integer in Oracle, or I'm missing something?

Reply:
in oracle - yes. I was trying to not break the logic if someone relies on default h2 regexp_replace logic on oracle mode

Reply:
We don't provide that level of compatibility for compatibility modes.